plugins/modules/redshift.py

```diff
@@ -254,15 +268,49 @@
             description: status of the enhanced vpc routing feature.
             returned: success
             type: bool
+        tags:
+            description: aws tags for cluster.
+            returned: success
+            type: dict
 '''
 
 try:
     import botocore
 except ImportError:
     pass  # caught by AnsibleAWSModule
 
-from ansible_collections.amazon.aws.plugins.module_utils.ec2 import AWSRetry, snake_dict_to_camel_dict
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import (
+    AWSRetry, snake_dict_to_camel_dict, boto3_tag_list_to_ansible_dict,
+    compare_aws_tags, ansible_dict_to_boto3_tag_list)
 from ansible_collections.amazon.aws.plugins.module_utils.aws.core import AnsibleAWSModule, is_boto3_error_code
+from ansible_collections.amazon.aws.plugins.module_utils.aws.iam import get_aws_account_id
+
+
+def _ensure_tags(redshift, identifier, existing_tags, module):
+    """Compares and update resource tags"""
+
+    account_id = get_aws_account_id(module)
+    region = module.params.get('region')
+    resource_arn = "arn:aws:redshift:{0}:{1}:cluster:{2}" .format(region, account_id, identifier)
```

Comment:
This makes an assumption that we're in the `aws` partition.  This doesn't hold true for China or GovCloud.

I've opened https://github.com/ansible-collections/amazon.aws/pull/136 which adds get_account_info which also returns the partition (using similar logic to get_account_id)

Reply:
Rather than holding this up indefinitely I think we come back and clean this up when we have the helper available to us.